game_state.py

```diff
@@ -146,10 +146,23 @@ def get_pawn(self, row, col, color):
     def check_cell(self, row, col):
         return -1 if row < 0 or row > (self.width - 1) or col < 0 or col > (self.width - 1) else self.board[row][col]
 
-    def draw_board(self):
-        for i in range(3):
-            print(self.board[i])
-        print()
+    def draw_board(self):        
+      print()
+      print(' ', end=' ')
+      blabels = [str(i) for i in range(3)]
```

Comment:
I suspect that many of these loops can be replaced with the `join` method on strings. For example, consider joining a list of integers with a vertical pipe:

```python
' | '.join([1, 2, 3]) == '1 | 2 | 3'
```

Can we simplify the logic in this function with this method?

Reply:
Yes, you are correct this method is a bit overcomplicated - I orginially used this to allow the board size to be changed dynamically in the future however I see this is unlikely to be required - I have replaced with a join of a simple list of strings.
`blabels = ['1', '2', '3']`
`print('|', ' | '.join(blabels), end=' ')`